six/modules/c++/six.sidd/tests/test_byte_swap.cpp

```diff
@@ -118,22 +118,13 @@ void write(const sys::Int16_T* data, bool useStream, bool byteSwap)
     }
 }
 
-void read(const std::string& filename, sys::Int16_T* data)
+void read(const std::string& filename, mem::ScopedArray<sys::Int16_T>& data)
 {
     six::NITFReadControl reader;
     reader.load(filename);
-    mem::SharedPtr<const six::Container> container = reader.getContainer();
-    const six::Data* const inData = container->getData(0);
 
-    const size_t numRows = inData->getNumRows();
-    const size_t numCols = inData->getNumCols();
     six::Region region;
-    region.setStartRow(0);
-    region.setStartCol(0);
-    region.setNumRows(numRows);
-    region.setNumCols(numCols);
-    region.setBuffer(reinterpret_cast<six::UByte*>(data));
-    reader.interleaved(region, 0);
+    reader.interleaved(region, 0, data);
```

Comment:
I'm still a little hung up on that we're not re-using the memory now... to me, just adding one more line (the `region.setBuffer()` line from before) and passing in a raw pointer here rather than the `ScopedArray` really is pretty legible code, but it is the case that we're talking a trivial amount of memory and this is a test program, so if you think this does improve readability, I'm good with it.